jax/_src/core.py

```diff
@@ -90,7 +90,9 @@ class Jaxpr:
   _effects: Effects
   _debug_info: JaxprDebugInfo | None
 
-  constvars = property(lambda self: self._constvars)
+  @property
+  def constvars(self) -> list[Var]:
+    return self._constvars
```

Comment:
MyPy cannot discern the type of the property otherwise.

An alternative solution would be to decorate this class with `@dataclass(frozen=True, slots=True)`, but dataclasses don't yet have converters (it's being discussed on Python-ideas), so the conversion to list of various arguments would have to be done on the caller side.

If we keep the expanded property, should I convert the rest of the properties?

Reply:
This (along with other similar properties) seems like a useful change that could be done in its own PR.